src/components/chat/ChatInput.tsx

```diff
@@ -0,0 +1,152 @@
+// src/components/chat/ChatInput.tsx
+import { useEffect, useRef, useState, useCallback } from 'react';
+import { Icon } from '@/components/Icon';
+
+type Props = {
+  onSend: (text: string) => Promise<void> | void;
+};
+
+export default function ChatInput({ onSend }: Props) {
+  const [value, setValue] = useState('');
+  const taRef = useRef<HTMLTextAreaElement>(null);
+
+  // 스펙
+  const BAR_MIN = 56; // 바 최소 높이
+  const W_EMPTY = 199; // 입력 전 너비(콘텐츠 폭)
+  const W_TYPED = 271; // 입력 시작 후 너비(콘텐츠 폭)
+  const ONE_H = 40; // 1줄 높이
+  const TWO_H = 60; // 2줄 높이(최대)
+  const H_PADX = 32; // 버블 좌우 padding 16+16 (px)
+
+  const [boxH, setBoxH] = useState<number>(ONE_H);
+  const hasText = value.trim().length > 0;
+
+  const autoResize = () => {
+    const el = taRef.current;
+    if (!el) return;
+
+    // 1) 현재 상태(199/271)에 맞춰 측정용 너비를 먼저 반영
+    const measureWidth = hasText ? W_TYPED : W_EMPTY;
+    el.style.width = `${measureWidth}px`;
+
+    // 2) 높이 초기화 후 실측
+    el.style.height = 'auto';
+
+    // 3) 1줄 임계값(라인하이트+패딩+보더)에 여유치(EPS) 포함
+    const cs = window.getComputedStyle(el);
+    const lineH = parseFloat(cs.lineHeight) || 20; // leading-5 기준 약 20px
+    const padV = (parseFloat(cs.paddingTop) || 0) + (parseFloat(cs.paddingBottom) || 0);
+    const borderV = (parseFloat(cs.borderTopWidth) || 0) + (parseFloat(cs.borderBottomWidth) || 0);
+    const EPS = 6; // 폰트/브라우저 오차 여유
+    const oneLineThreshold = lineH + padV + borderV + EPS;
+
+    const contentH = el.scrollHeight;
+
+    // 4) 실제로 줄바꿈이 생겼을 때만 60으로 확대
+    const target = contentH > oneLineThreshold ? TWO_H : ONE_H;
+    setBoxH(target);
+
+    // 5) textarea 자체 높이/최대높이 + 스크롤 처리
+    el.style.height = `${Math.min(contentH, TWO_H)}px`;
+    el.style.maxHeight = `${TWO_H}px`;
+    el.style.overflowY = contentH > TWO_H ? 'auto' : 'hidden';
+  };
+
+  useEffect(() => {
+    autoResize();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [value]); // hasText는 value 변화에 종속
+
+  const send = useCallback(async () => {
+    const text = value.trim();
+    if (!text) return;
+    await onSend(text);
+    setValue('');
+    taRef.current?.focus();
+  }, [onSend, value]);
+
+  const onKeyDown: React.KeyboardEventHandler<HTMLTextAreaElement> = (e) => {
+    if (e.nativeEvent.isComposing) return;
+    if (e.key === 'Enter' && !e.shiftKey) {
+      e.preventDefault();
+      void send();
+    }
+  };
+
+  return (
+    <div
+      className="flex items-center gap-3 bg-[var(--white)] px-3"
+      style={{ minHeight: BAR_MIN, height: Math.max(BAR_MIN, boxH) }}
```

Comment:
```suggestion
      style={{ minHeight: BAR_MIN, height: Math.max(BAR_MIN, boxH+20) }}
```
이렇게 하면 위에 패딩이 안 뭉개지고 그대로 남아있게 되는거 같습니다!
지금 배포 링크에서는 input box로 되어있어서 textarea + 반응형되어있는 로컬로 돌려봤는데, 괜찮은 거 같아요!
<img width="491" height="155" alt="image" src="https://github.com/user-attachments/assets/c6d4393b-b1d3-4a06-8bcb-29958d3b872d" />